app/types/scheduler.py

```diff
@@ -0,0 +1,229 @@
+import asyncio
+import logging
+from collections.abc import AsyncGenerator, Callable, Coroutine
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any
+
+from arq.connections import RedisSettings
+from arq.jobs import Job
+from arq.typing import WorkerSettingsBase
+from arq.worker import create_worker
+from sqlalchemy.ext.asyncio import AsyncSession
+
+if TYPE_CHECKING:
+    from arq import Worker
+
+scheduler_logger = logging.getLogger("scheduler")
+
+
+async def run_task(
+    ctx,
+    job_function: Callable[..., Coroutine[Any, Any, Any]],
+    _get_db: Callable[[], AsyncGenerator[AsyncSession, None]],
+    **kwargs,
+):
+    scheduler_logger.debug(f"Job function consumed {job_function}")
+
+    if "db" in kwargs:
+        async for db in _get_db():
+            kwargs["db"] = db
+
+    return await job_function(**kwargs)
+
+
+class Scheduler:
+    """
+    An [arq](https://arq-docs.helpmanual.io/) scheduler.
+    The wrapper is intended to be used inside a FastAPI worker.
+
+    The scheduler use a Redis database to store planned jobs.
+    """
+
+    # See https://github.com/fastapi/fastapi/discussions/9143#discussioncomment-5157572
+
+    def __init__(self):
+        # ArqWorker, in charge of scheduling and executing tasks
+        self.worker: Worker | None = None
+        # Task will contain the asyncio task that runs the worker
+        self.task: asyncio.Task | None = None
+        # Pointer to the get_db dependency
+        # self._get_db: Callable[[], AsyncGenerator[AsyncSession, None]]
+
+    async def start(
+        self,
+        redis_host: str,
+        redis_port: int,
+        redis_password: str | None,
+        _get_db: Callable[[], AsyncGenerator[AsyncSession, None]],
+        **kwargs,
+    ):
+        """
+        Parameters:
+        - redis_host: str
+        - redis_port: int
+        - redis_password: str
+        - get_db: Callable[[], AsyncGenerator[AsyncSession, None]] a pointer to `get_db` dependency
+        """
+
+        class ArqWorkerSettings(WorkerSettingsBase):
+            functions = [run_task]
+            allow_abort_jobs = True
+            keep_result_forever = True
+            redis_settings = RedisSettings(
+                host=redis_host,
+                port=redis_port,
+                password=redis_password if redis_password is not None else "",
```

Comment:
Does this works if we pass `""` as password?

Reply:
Empty password is allowed depending if redis has no password set up

Reply:
Brilliant, thanks
```suggestion
                password=redis_password or "",
```